lib/fluent/plugin/in_watch_process.rb

```diff
@@ -44,53 +42,237 @@ def shutdown
 
-      def OS.mac?
-       (/darwin/ =~ RUBY_PLATFORM) != nil
+          data
+        end
       end
+      private_constant :BaseUnixLike
+
+      class Linux < BaseUnixLike
+        DEFAULT_COMMAND = "LANG=en_US.UTF-8 && ps -ewwo lstart,user:20,pid,ppid,time,%cpu,%mem,rss,sz,s,comm,cmd"
 
-      def OS.unix?
-        !OS.windows?
+        def initialize(keys, command)
+          super(keys, command)
+          @command = @command || DEFAULT_COMMAND
+        end
+      end
+
+      class Mac < BaseUnixLike
+        DEFAULT_COMMAND = "LANG=en_US.UTF-8 && ps -ewwo lstart,user,pid,ppid,time,%cpu,%mem,rss,vsz,state,comm,command"
+
+        def initialize(keys, command)
+          super(keys, command)
+          @command = @command || DEFAULT_COMMAND
+        end
       end
 
-      def OS.linux?
-        OS.unix? and not OS.mac?
+      class Windows < Base
+        # Values are from the "System.Diagnostics.Process" object properties that can be taken by the "Get-Process" command.
+        # You can check the all properties by the "(Get-Process)[0] | Get-Member" command.
+        DEFAULT_PARAMS = {
+          "start_time" => "StartTime",
+          "user" => "UserName",
+          "sid" => "SessionId",
+          "pid" => "Id",
+          "cpu_second" => "CPU",
+          "working_set" => "WorkingSet",
+          "virtual_memory_size" => "VirtualMemorySize",
+          "handles" => "HandleCount",
+          "proc_name" => "ProcessName",
+        }
+
+        def initialize(keys, command)
+          super(keys, command)
+          @keys = @keys || DEFAULT_PARAMS.keys
+          @command = @command || default_command
+        end
+
+        def default_types
+          "sid:integer,pid:integer,cpu_second:float,working_set:integer,virtual_memory_size:integer,handles:integer"
+        end
+
+        def parse_line(line)
+          values = line.chomp.strip.parse_csv.map { |e| e ? e : "" }
+          Hash[@keys.zip(values)]
+        end
+
+        def default_command
+          command = [
+            command_ps,
+            pipe_filtering_normal_ps,
+            pipe_select_columns,
+            pipe_fixing_locale,
+            pipe_formatting_output,
+          ].join
+          "powershell -command \"#{command}\""
+        end
+
+        private
+
+        def command_ps
+          if @keys.include?("user")
+            # The "IncludeUserName" option is needed to get the username, but this option requires Administrator privilege.
+            "Get-Process -IncludeUserName"
+          else
+            "Get-Process"
+          end
+        end
+
+        def pipe_filtering_normal_ps
+          # There are some special processes that don't have some properties, such as the "Idle" process.
+          # Normally, these are specific to the system and are not important, so exclude them.
+          # Note: The same situation can occur in some processes if there are no Administrator privilege.
```

Comment:
Because I'm not familiar with PowerShell, such comment is much helpful for me :+1: